### PR TITLE
CLOUDP-85749: Add new available region endpoint to the atlas go client

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

I have added the service to handle the endpoint `GET /api/atlas/v1.0/groups/{GROUP-ID}/clusters/provider/regions`

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: mongodbatlas/cloud_provider_regions.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	Links      []*Link          `json:"links,omitempty"`
+	Results    []*CloudProvider `json:"results,omitempty"`
+	TotalCount int              `json:"totalCount,omitempty"`
+}
+
+// CloudProviderRegionsOptions specifies the optional parameters to the CloudProviderRegions Get method.
+type CloudProviderRegionsOptions struct {
+	Provider           string `url:"provider,omitempty"`
+	Tier               string `url:"tier,omitempty"`
+	CrossCloudProvider bool   `url:"maxDate"`
```
Comment (gssbzn) [inline mongodbatlas/cloud_provider_regions.go:68]: copy/paste?
```suggestion
	CrossCloudProvider bool   `url:"crossCloudProvider"`
```

**Review by gssbzn — CHANGES_REQUESTED** [review]
just one small typo

**Review by gssbzn — APPROVED** [review]
LGTM cc @themantissa

**Review by themantissa — APPROVED** [review]
LGTM! 😄  fyi @leofigy @pitakill @coderGo93 we'll be able to make use of this in the future to get a list of regions we have available.

**Pull request merged**